test/sqllogictest/column_knowledge.slt

```diff
@@ -379,24 +360,21 @@ EXPLAIN SELECT * FROM t1 WHERE 123 = (SELECT t2.f1 FROM t2);
 %1 =
 | Get materialize.public.t2 (u3)
 | Filter (#0 = 123)
-| Project (#0)
+| Project ()
 
 %2 =
 | Get materialize.public.t2 (u3)
 | Reduce group=()
 | | agg count(true)
 | Filter (err: more than one record produced in subquery), (#0 > 1)
-| Map (err: more than one record produced in subquery)
-| Project (#1)
+| Project ()
```

Comment:
I wonder if this sort of changes means that our subquery error code might be at risk (just bolting on an error field might not be enough if you don't look at that field). That stuff needs to be rethought anyhow, so no worries for this PR.

Reply:
I manually tested this query, and it still behaves correctly. It returns all the rows in t1 if t2 has one or fewer rows and there is an evaluation error otherwise.